src/App.tsx

```diff
@@ -26,70 +27,107 @@ type ReorderOptions = {
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  setSortType = (sortType: SortType) => {
+    this.setState({ sortType });
+  };
+
+  setReverse = () => {
+    this.setState((Prev: State) => ({ isReversed: !Prev.isReversed }));
+  };
+
+  render() {
+    const { sortType, isReversed }: ReorderOptions = this.state;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button',
+              'is-info',
+              { 'is-light': sortType !== SortType.ALPHABET })}
+            onClick={() => this.setSortType(SortType.ALPHABET)}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button',
+              'is-info',
+              { 'is-light': sortType !== SortType.LENGTH })}
+            onClick={() => this.setSortType(SortType.LENGTH)}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button',
+              'is-info', { 'is-light': isReversed === false })}
+            onClick={() => this.setReverse()}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== 0 || isReversed) && (
```

Comment:
Nitpick: When the logic for rendering consist of multiple logical operators, you could lift it above return statement. e.g.
`const isResetVisible = (sortType !== 0 || isReversed)`
Then there is less clutter in JSX